primitives/chain-polkadot-bulletin/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Polkadot Bulletin Chain primitives.
+
+#![warn(missing_docs)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use bp_header_chain::ChainWithGrandpa;
+use bp_messages::{ChainWithMessages, MessageNonce};
+use bp_runtime::{
+	decl_bridge_finality_runtime_apis, decl_bridge_messages_runtime_apis, Chain, ChainId,
+};
+use frame_support::{
+	dispatch::DispatchClass,
+	parameter_types,
+	weights::{constants::WEIGHT_REF_TIME_PER_SECOND, Weight},
+};
+use frame_system::limits;
+use sp_runtime::{Perbill, StateVersion};
+
+// This chain reuses most of Polkadot primitives.
```

Comment:
Not sure it's a good idea to tie these things to Polkadot? If they are changed in Polkadot they're not going to automatically change in the Bulletin Chain...

Reply:
Should just take the definitions from `runtime/src/lib.rs` in the Bulletin Chain repo?

Reply:
> Not sure it's a good idea to tie these things to Polkadot? If they are changed in Polkadot they're not going to automatically change in the Bulletin Chain...

True, but if types will change in Polkadot, they won't automatically change in bridges `bp-polkadot-core` crate as well :) Because polkadot uses its own crates and our `bp-polkadot-core` is just a minimal knowledge of Polkadot-like chains that we need to implement bridges. We need to track the possible type changes and act accordingly.

In bridges there are two "base" chain types that we are able to handle currently - standalone (Polkadot-like) chains and parachains (Cumulus-based mostly). To me the Bulletin Chain looks more Polkadot-like, hence the usage of the `bp_polkadot_core` crate. I could just copy paste those types and constants here, but it'll just increase amount of code and won't change anything in practice :) But feel free to insist - it isn't a big deal

> Should just take the definitions from runtime/src/lib.rs in the Bulletin Chain repo?

There are two huge issues with that:
- we are in a different repo and if we'll start referring bulleting chain crates here, it'll cause circular references;
- this crate will be used by polkadot bridge hub runtime and we don't want to embed (at least build) another runtime when building Polkadot BH.

Actually, with your chain it is easier and I'd like to "use typedefs from this crate in Bulletin Chain runtime to avoid code duplication" as the PR description states later, if you'll accept such PR. I.e. instead of having `type BlockNumber = u32;` in bulletin chain runtime, I'd like to have it here and then simply reuse in runtime. WDYT?

Reply:
I meant cut the definitions out of the Bulletin Chain's `runtime/src/lib.rs` and paste them here. They're just typedefs of Substrate types AFAIK, shouldn't be any dependency issues.

I'm happy to have the Bulletin Chain then get the definitions from here to avoid duplication; we're going to be depending on this stuff anyway.